src/VisualStudio/IntegrationTest/TestUtilities/Interop/AutomationRetry/AutomationRetryWrapper`1.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.Interop.AutomationRetry
+{
+    internal abstract class AutomationRetryWrapper<T> : IRetryWrapper, ICustomQueryInterface
+    {
+        internal const int UIA_E_ELEMENTNOTAVAILABLE = unchecked((int)0x80040201);
+
+        /// <summary>
+        /// The number of times to retry a UI automation operation that failed with
+        /// <see cref="UIA_E_ELEMENTNOTAVAILABLE"/>, not counting the initial call. A value of 9 means the operation
+        /// will be attempted a total of ten times.
+        /// </summary>
+        private const int AutomationRetryCount = 9;
+
+        /// <summary>
+        /// The delay between retrying a UI automation operation that failed with
+        /// <see cref="UIA_E_ELEMENTNOTAVAILABLE"/>.
+        /// </summary>
+        private static readonly TimeSpan AutomationRetryDelay = TimeSpan.FromMilliseconds(100);
+
+        private static readonly Guid IID_IManagedObject = new Guid("C3FCC19E-A970-11D2-8B5A-00A0C9B7C9C4");
+
+        protected AutomationRetryWrapper(T automationObject)
+        {
+            AutomationObject = automationObject;
+            var comCallableWrapper = Marshal.GetIUnknownForObject(automationObject);
```

Comment:
Cleaning up some of these names would be good.

I don't think this is a CCW, I think this is the native handle.